modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/discovery/impl/DiscoveryServiceImpl.java

```diff
@@ -109,4 +100,37 @@ public Response discoverGet() {
         }
         return Response.ok().build();
     }
+
+    /**
+     * Get authentication policy from the tenant configuration, otherwise set default value as Federated.
+     *
+     * @return Authentication policy.
+     * @throws WindowsDeviceEnrolmentException
+     */
+    public String getAuthPolicy() throws WindowsDeviceEnrolmentException {
+        String authPolicy = null;
+        List<ConfigurationEntry> tenantConfigurations;
+        try {
+            if (WindowsAPIUtils.getTenantConfigurationData() != null) {
+                tenantConfigurations = WindowsAPIUtils.getTenantConfigurationData();
```

Comment:
rather than calling WindowsAPIUtils.getTenantConfiguration() two times, better to call it once, get the value to a local variable and then check for null value.